engine/src/main/java/com/datatorrent/stram/stream/BufferServerSubscriber.java

```diff
@@ -242,6 +243,21 @@ public long getByteCount(boolean reset)
     }
 
     @Override
+    public int size(final boolean dataTupleAware)
+    {
+      int size = size();
+      if (dataTupleAware) {
```

Comment:
@gauravgopi123 How critical is to filter out control tuples for dynamic partitioning? Is there a check against zero data tuple aware queue size?